lib/ruby_smb/smb1/packet/trans2/data_block.rb

```diff
@@ -35,7 +35,7 @@ def pad1_length
           # Determines the correct length for the padding in front of
           # #trans2_data. It should always force a 4-byte alignment.
           def pad2_length
-            if enable_padding
+            if enable_padding && trans2_data.num_bytes > 0
```

Comment:
Fixed in da991069312f1b2a231778af43a6aea209a75aa2. I when through all the Trans2 responses that include a raw buffer (because the contents are determined by the accompanying request which isn't part of the data structure) and fixed them all to be proper strings with a new `#buffer_read_length` common method. I also updated all of the specs to ensure that they're padded when necessary and can parse their binary representations.

Reply:
Thanks! That's great!